### PR TITLE
Pull Request for Issue1167: Make the report file date in front of time

### PR DESCRIPTION
**Pull request by XuetaoLiu-CLS**
https://github.com/acquaman/acquaman/issues/1167

1) the reports can be easier to be found since they are ordered by date and then time.
2) add year to the subfolder of the beamline folder, so that the history reports will be organized nicely

@dretrex @davidChevrier

### CODE REVIEW

File: source/application/AMCrashReporter.cpp [hunk 1]
```diff
@@ -118,6 +118,12 @@ AMCrashMonitor::AMCrashMonitor(const QString &executableFullPath, const QString
 	errorFilePath_ = errorFilePath;
 	watchingPID_ = watchingPID;
 
+	//check the current year directory exist or not, if not create one.
+	QString targetPath = QString("%1/%2").arg(errorFilePath_).arg(QDateTime::currentDateTime().toString("yyyy"));
+	if (!QDir(targetPath).exists()) {
+		QDir().mkdir(targetPath);
```
Comment (davidChevrier) [inline source/application/AMCrashReporter.cpp]: It's possible for this operation to fail. You might want to handle that situation.

Comment (XuetaoLiu-CLS) [inline source/application/AMCrashReporter.cpp]: @davidChevrier I deleted these codes and move them down.

**Comment by davidChevrier** [comment]
Looks much better. Have you tested this in the wild? Just checking before I pull it into master.

**Comment by XuetaoLiu-CLS** [comment]
No, I didn't test it yet since SXRMB beamline is occupied now. But I do tested the codes for directory checking and creating code in a seperate QT project. and this is the major changes actually.

**Comment by davidChevrier** [comment]
Okay. Let's get this tested next Monday on SXRMB. That way we can make sure that everything is good to go before it winds up in master.

**Comment by XuetaoLiu-CLS** [comment]
Test is done on SXRMB beamline.
On SXRMB beamline, the AMCrashReporter is not launched by the AMCrashMonitor automatically when crash happened by some unknown reason. This is known issue and need to be investigated.

When launching AMCrashReporter manually, it will create the year subfolder under the beamline folder and create crash file as expected.

**Pull request merged**